### ISSUE TITLE
Can't Open Old ORA files after upgrade

### ISSUE THREAD

**Issue by DrawMeister123** [mypaint/mypaint#585]
I get this error message when attempting to open old ORA files in the latest version of MYPAINT

gdk-pixbuf-error-quark: Fatal error reading PNG image file: Decompression error in IDAT (0)

Any help appreciated, thanks.

Split off from #582

**Comment by DrawMeister123** [comment]
PS if you need more information please ELI5 how to get it. NOT emphasis NOT expert - I use the stuff only

**Comment by odysseywestra** [comment]
Is this only one file that is causing this, or is it multiple files? Would you be able to provide the file for us to download it so we can test it? Also are you using the w32 version or w64 version of MyPaint from our [Release Page](https://github.com/mypaint/mypaint/releases/tag/v1.2.0)?

Also could you provide us with the full error message please. Here what an example of what it looks like: 

```python
Mypaint version: 1.3.0-alpha+git.d0deedb
System information: Linux-4.2.0-25-generic-x86_64-with-Ubuntu-15.10-wily
Traceback (most recent call last):
  File "/usr/local/share/mypaint/gui/application.py", line 802, crash_program_cb(self=<gui.application.Application object>, action=<Action object at 0x7f68c8f2c5f0 (GtkAction at 0x22e91b0)>)
            """Tests exception handling."""
            raise Exception("This is a crash caused by the user.")
  variables: {'Exception': ('builtin', <type 'exceptions.Exception'>)}
Exception: This is a crash caused by the user.
```
That information will help us greatly pinpoint where in MyPaint's code is causing the problem.

**Comment by achadwick** [comment]
That message appear in a popup I think, so a screengrab of the dialog might be OK.

**Comment by achadwick** [comment]
@DrawMeister123
Just to confirm, is this MyPaint 1.2.0 or a beta (prerelease)?

It may be that the file is actually corrupt. Let's try to rule that out. Can it be opened in the latest release of GIMP?

**Comment by DrawMeister123** [comment]
Sorry no idea how to get that error message information, the message is in a popup which I copy/pasted exactly so screen grab isn't going to tell you anything different. 

I'm using the 32bit version - the latest stable Win32 download available from the site. 

The file isn't corrupt, and it isn't just one file - it won't open any of my old ORA's. I tested that possibility.

**Comment by DrawMeister123** [comment]
Just to say that I have reverted to 1.1 which works. Every heard the expression - if ain't broke don't fix it! This applies to windows in the extreme. Tried to see if there was an updated driver for my tablet. Thought I found one. Ended up with a black screen. Had to open in safe mode, uninstall the driver, reboot. Reinstall tablet driver that I know does work. Reboot. In the process decided old but working Mypaint better than no Mypaint.

**Comment by DrawMeister123** [comment]
FYI this right here, this is THE problem with Open Source. I am huge fan, used Open Source stuff for years, my bad for using a new release even if labeled 'stable'. When there is a problem with new stuff, there is little or no support, and you are basically supposed to fix it yourself. Not all users are programmers, some of us just like to have to stuff that works, preferably out the box, without problems, and if there is a problem, some actual help solving it.

**Comment by achadwick** [comment]
Well, okay... I'm truly sorry you feel that way. No, really. We do need more eyes on this (eyes, and time...).

Have you tried asking on the MyPaint Community Forums <http://community.mypaint.org/> already? Looks like not. We try to keep the forums fairly friendly and non-techie, and it's intended as a bridge between artists and developers for when the technical stuff gets TOO INTENSE (...!  But yeah, that is a failing. We do what we can to make community and support not be totally awful.)

My interest here is finding out whether this is something we can fix for everyone. Are you OK with cracking open one of your old `.ORA` files and attaching one of the `.PNG` files in it in the forums or here? They are renamed `.ZIP` archives (lots of files and folders in one), and I can walk you through how to examine its guts in Windows.

**Comment by achadwick** [comment]
Made a forum thread about it: http://community.mypaint.org/t/windows-users-any-problems-opening-old-ora-files-with-v1-2-0/165

**Comment by achadwick** [comment]
@DrawMeister123 
Silly question, just want to rule out another possibility. Is there any possibility that these files were written by another program, like an old GIMP OpenRaster plugin?

**Comment by DrawMeister123** [comment]
Although I have GIMP I have not used it to open ORA's. All my painting is done exclusively in Mypaint. I have attached one of my "old" files for you. 

Ok that don't work. Hmmm ... 

https://drive.google.com/file/d/0B0cGKCKvN81jRmtKckR0TEVCYUE/view?usp=sharing

uploaded to google drive.

**Comment by zb13y** [comment]
I can open it. Windows 10 x64, 1.2.0. I have old oras from 1.0, i can open them too.

**Comment by gwojcik** [comment]
I can reproduce this bug on Vista SP1 32 using MyPaint 1.2 with file from @DrawMeister123.
Png files extracted from ora file can be opened in MyPaint without error.

**Comment by Itzcuauhtli** [comment]
You try to open your .ora files in a different program, like GIMP or Krita? Because if these file open without problem, the problem is from a MyPaint's bug. But if the files can't open, then the problem comes from your own files that could be damaged.

A temporaly solution is if your files open with GIMP or Krita, re-save the files with this programs or in a new archive

**Comment by odysseywestra** [comment]
Though that may be an accepting workaround, there is still the issue of
this bug. Right now it seems to only affect certain Windows installations
cause I am able to open it on my Linux Rig. Not sure if Mac OSX is affected
by this.

On Wed, Feb 3, 2016, 3:42 PM Itz <notifications@github.com> wrote:

> You try to open your .ora files in a different program, like GIMP or
> Krita? Because if these file open without problem, the problem is from a
> MyPaint's bug. But if the files can't open, then the problem comes from
> your own files that could be damaged.
>
> A temporaly solution is if your files open with GIMP or Krita, re-save the
> files with this programs or in a new archive
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mypaint/mypaint/issues/585#issuecomment-179534276>.
>
-- 

-- Albert Westra

**Comment by KirshD** [comment]
I might have figured out something, I've had some files refusing to load and I've noticed a pattern. I switched from 1.2.0beta.1 x64 to 1.2.0 final x86. The files that are refusing to load entirely are ones where I moved a layer's position with the move layer tool then saved. Note the "entirely", a couple of my images with an unmoved upper layers would load the layers but none of the moved underlying layers. Oddly though, I opened one of my corrupted images in a copy of the beta version again and just saved it again with no edits, and now I can open it again in 1.2.0 final. Is there a relative position being saved for layers that might be interfering with the data when loaded?

**Comment by achadwick** [comment]
@KirshD layers all get the same coordinate space, sort of. Repositioning a layer chops up existing tiles and reallocates the slices to new tiles. But the new ones are on the same grid as every other layer, same as before the move. Same save routine, same everything. What you're seeing is weird, and most likely a different bug.

**Comment by calsioro** [comment]
Build 1.2.0+git.f62444e.dirty x64, windows 7. OP's file opens fine for me.

**Comment by achadwick** [comment]
@DrawMeister123 
[MyPaint v1.2.1](https://github.com/mypaint/mypaint/releases/tag/v1.2.1) was released recently. Please could you test it out, and let us know whether the bug is still happening?

For what it's worth, I can install it so that double-clicking the file opens it up just fine in both the w32 and w64 builds.

I think this issue should be considered unreproducable.

**Comment by DrawMeister123** [comment]
Thanks. I will download the new version, although upgrading to Win 10 also seems to have resolved the issue.

**Comment by achadwick** [comment]
@DrawMeister123 
Thanks for the update. I'll consider this one as non-reproducible.

@KirshD 
Not sure what's going on here. Please let us know if it's still happening (in a fresh issue please!)
